spec.md

```diff
@@ -729,21 +729,23 @@ Procedure:
 
 8. if X1 != X2 return INVALID
 
-9. c = H(Abar || A' || h0 || C1 || d || h0 || h\[i1\] || ... || h\[iR\] || C2 || nonce)
+9. c_v = H(Abar || A' || h0 || C1 || d || h0 || h\[i1\] || ... || h\[iR\] || C2 || nonce)
 
-10. T1 = Abar - d
+10. if c != c_v return INVALID
 
-11. T2 = P1 + h\[i1\] \* Rmsg\[1\] + ... + h\[iR\] \* Rmsg\[R\]
+11. T1 = Abar - d
```

Comment:
These steps are not necessary since they should be handled by the challenge hash.

Reply:
I.E 11-16 are not necessary

Reply:
See https://hackmd.io/Q587Q9p7T5ab30NTn4MvTA#Change-selective-disclosure-proofs-to-just-be-Schnorr-Proofs-and-Fiat-Shamir

Reply:
Thank you for the link. It is very interesting. Perhaps I will have to think more about it but it seems to me that some of the steps are needed. If we change the hash calculated at step 9 here to contain Y1 and Y2 instead of C1 and C2 (and of course to be calculated in a later step), perhaps we could remove the checks at steps 15 and 16, and consequently remove C1 and C2 from the spk returned from SpkGen. However, steps 10 to 14 don't seem to me that we could remove them, they are an integral part of the Schnorr proof after all.

Reply:
The problem is it’s being calculated twice. Once for the hash and then again checked for equality which is why I said in my write up they can be removed

Reply:
If the values are different, the hash will be too.